### PR TITLE
misc: Prevent builtins on transient functions

### PR DESCRIPTION
**Pull request by FabioLuporini**

### CODE REVIEW

File: devito/builtins/utils.py [hunk 1]
```diff
@@ -131,3 +132,23 @@ def wrapper(*args, **kwargs):
...
+
+def check_args(func):
+    """
+    Perform checks on the arguments supplied to a builtin.
+    """
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        for i in args:
+            try:
```
Comment (mloubout) [inline devito/builtins/utils.py:145]: Should only be an issue on Device platform, CPU transient do hold the data

File: devito/builtins/utils.py [hunk 2]
```diff
@@ -131,3 +132,23 @@ def wrapper(*args, **kwargs):
         return func(*processed, argmap=argmap, **kwargs)
 
     return wrapper
+
+
+def check_args(func):
```
Comment (mloubout) [inline devito/builtins/utils.py:137]: check_builtins_args

**Review by georgebisbas — APPROVED** [review]

File: devito/builtins/utils.py (hunk 2 above)
Comment (FabioLuporini) [reply devito/builtins/utils.py:137]: OK

File: devito/builtins/utils.py (hunk 1 above)
Comment (FabioLuporini) [reply devito/builtins/utils.py:145]: OK, tweaking

File: tests/test_builtins.py [hunk 3]
```diff
@@ -518,3 +518,11 @@ def test_issue_1863(self):
         assert type(v1) is np.int32
         assert type(v2) is np.float32
         assert type(v3) is np.float64
+
+    def test_is_transient(self):
```
[the remark this answers is not in the document]
Comment (FabioLuporini) [reply tests/test_builtins.py:522]: OK

[the remark this answers is not in the document]
Comment (FabioLuporini) [reply tests/test_builtins.py:522]: moving to `test_gpu_common` otherwise CI won't pick it up

**Review by mloubout — APPROVED** [review]

**Pull request merged**